### PR TITLE
Speed improv & bug fixes

### PR DESCRIPTION
**Pull request by flanfly**

### CODE REVIEW

File: lib/src/loader.rs [hunk 1]
```diff
@@ -218,9 +218,9 @@ fn load_elf(bytes: &[u8], name: String) -> Result<(Project,Machine)> {
 
     // for now we comment adding symbols from strippable symbol table:
     // we don't have an easy way/API to check if duplicate symbol/function targest have been added
-    // for sym in &binary.syms {
-    //     add_sym(&mut prog, sym, &binary.strtab);
-    // }
+     for sym in &binary.syms {
+         add_sym(&mut prog, sym, &binary.strtab);
+     }
```
Comment (m4b) [inline lib/src/loader.rs]: Does the function table dedup automatically now ? I.e., what changed such that the above comment doesn't apply?

**Comment by flanfly** [comment]
@m4b does that fix it?

**Comment by flanfly** [comment]
Ok, but the symbol value must be the same, right?

**Comment by m4b** [comment]
Ya the symbol address would have to be the same. That actually might be a better way to do it.

**Pull request merged**